src/main/java/hudson/remoting/Request.java

```diff
@@ -234,11 +234,14 @@ public boolean isDone() {
             public RSP get() throws InterruptedException, ExecutionException {
                 synchronized(Request.this) {
                     try {
-                        while(response==null) {
+                        while(response==null && !channel.isInClosed()) {
                             if (isCancelled()) {
                                 throw new CancellationException();
                             }
-                            Request.this.wait(); // wait until the response arrives
+                            Request.this.wait(30*1000); // wait until the response arrives
```

Comment:
This is the exact same timeout for recheck the while condition as is used in `call()` hence using that timeout.

Spin-looping once every 30 seconds is acceptable to detect channel closed conditions. Longer timeouts do not seem to make sense, and shorter ones will just generate more loop. If you want to parameterize then all 3 uses of `30*1000` should be parameterized... i.e. a separate change from this